ksqldb-engine/src/main/java/io/confluent/ksql/query/QueryRegistryImpl.java

```diff
@@ -272,6 +273,7 @@ public PersistentQueryMetadata createOrReplacePersistentQuery(
 
     if (sharedRuntimeId.isPresent() && (oldQuery == null
         || oldQuery instanceof BinPackedPersistentQueryMetadataImpl)) {
+      throwOnNonQueryLevelConfigs(config.getOverrides());
```

Comment:
Sorry for the late feedback here. We should only do validation like this in the validation phase of executing a statement. Otherwise we might change the validation later on in a way that causes us to fail statements we've already accepted. The pattern for doing that is to do this type of validation in the sandbox instance only - can you make that change in a follow-up?

Reply:
https://github.com/confluentinc/ksql/pull/8722 sure

Those configs will be ignored but it won't mean there is a record in the cmd topic that will cause it to fail.